qiskit_nature/operators/second_quantization/star_algebra.py

```diff
@@ -0,0 +1,86 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Sum Operator base interface."""
+
+from abc import ABC, abstractmethod
+
+from .multiply import MultiplyMixin
+from .tolerances import TolerancesMixin
+
+
+class StarAlgebraMixin(MultiplyMixin, TolerancesMixin, ABC):
+    """The star algebra mixin class.
+
+    Star algebra means an algebra with adjoint (dagger).
+    """
+
+    @abstractmethod
+    def _add(self, other):
+        r""" Return Operator addition of self and other, overloaded by ``+``.
+        """
+        raise NotImplementedError
+
+    def __add__(self, other):
+        return self._add(other)
+
+    def __sub__(self, other):
+        return self._add(other._multiply(-1))
```

Comment:
Would it be preferable to have an _sub method defined, like _add, but where is has the implementation here so you do not need to override by default. But if you had a better way to do sub then you could. Any why would you not say `other * -1` or even `-other` rather than directly calling upon some method here to do it.

Reply:
Indeed, if you want to override `__sub__`, it is better to prepare `_sub` and override it. Adding a method does not involve deprecation, so it is easy to do even after release.

> Any why would you not say other * -1 or even -other rather than directly calling upon some method here to do it.

There is no deep reason for this, but I feel that it is better to have a shallow hierarchy of calls as possible.

Reply:
So far I cannot think of a reasonable scenario in which the existence of `_sub` would be useful. And even if it does occur, as @ikkoham has pointed out, we can still add it later on, if needed.
However, I would like to see `-other` here as I think it easier to read. And as you point out yourself, it should not change the performance :+1: 
```suggestion
        return self._add(-other)
```

PS: this is also what the `__sub__` method looks like in the currently open PR of Terra.